lib/android/app/src/main/java/com/reactnativenavigation/views/TitleBarButton.java

```diff
@@ -0,0 +1,163 @@
+package com.reactnativenavigation.views;
+
+import android.content.Context;
+import android.graphics.Color;
+import android.graphics.drawable.Drawable;
+import android.support.annotation.NonNull;
+import android.support.v7.widget.Toolbar;
+import android.text.Spannable;
+import android.text.SpannableString;
+import android.text.style.AbsoluteSizeSpan;
+import android.text.style.RelativeSizeSpan;
+import android.text.style.TextAppearanceSpan;
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.TextView;
+
+import com.reactnativenavigation.parse.Button;
+import com.reactnativenavigation.parse.NavigationOptions;
+import com.reactnativenavigation.utils.ImageUtils;
+import com.reactnativenavigation.utils.UiUtils;
+import com.reactnativenavigation.viewcontrollers.ContainerViewController;
+
+import java.util.ArrayList;
+
+public class TitleBarButton implements MenuItem.OnMenuItemClickListener {
+	private Toolbar toolbar;
+	private final Button button;
+	private ContainerViewController.ContainerView containerView;
+	private Drawable icon;
+
+	TitleBarButton(ContainerViewController.ContainerView containerView, Toolbar toolbar, Button button) {
+		this.containerView = containerView;
+		this.toolbar = toolbar;
+		this.button = button;
+	}
+
+	public void addToMenu(Context context, final Menu menu) {
+		MenuItem menuItem = menu.add(button.title);
+		menuItem.setShowAsAction(button.showAsAction);
+		menuItem.setEnabled(button.disabled != NavigationOptions.BooleanOptions.True);
+		menuItem.setOnMenuItemClickListener(this);
+
+		if (hasIcon()) {
+			setIcon(context, menuItem);
+		} else {
+			setTextColor();
+			setFontSize(menuItem);
+		}
+	}
+
+	public void setNavigationIcon(Context context) {
```

Comment:
please, rename to "applyNavigationIcon". Now it's kind of misleading, 'cause it looks like you need to provide icon, that should be set